src/merge-async-iterables.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Merges an iterable on top of another iterable.
+ *
+ * The second iterable can support the concept of delete entries (tombstones)
+ * and if the value of the second iterable is a delete value then the value is
+ * not included in the resulting iterable.
+ */
+export async function* mergeAsyncIterables<A, B>(
```

Comment:
I see what you're saying. If you prefer to expose this and have callers use it that would be fine with me.